cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp

```diff
@@ -77,22 +77,37 @@ struct get_core_wide_kernel {
                     count_type count = 0;
                     for (std::int64_t j = 0; j < row_count; j++) {
                         Float sum = Float(0);
-                        for (std::int64_t i = local_id; i < column_count; i += local_size) {
+                        for (std::int64_t i = local_id; i < column_count / 2; i += local_size) {
+                            Float val = data_ptr[(block_start + wg_id) * column_count + i] -
+                                        data_ptr[j * column_count + i];
+                            sum += val * val;
+                        }
+                        Float distance_check =
+                            sycl::reduce_over_group(sg, sum, sycl::ext::oneapi::plus<Float>());
```

Comment:
how expensive is this distance check? would it make sense to add inside the for loop and break immediately once threshold is achieved instead of break for loop into 2?

Reply:
Its a good point. I did it mostly based on results on real data, and looks like its expensive to call this reduce each loop iteration. May be its better to make some mechanism to count number of splits based on column count

Reply:
Additional question to this line: you are reducing over subgroup (passing sg as first arg), however computations are over group. Does it affect on accuracy or  everything is fine?

Reply:
All tests are passed, and DBI is the same as in master, but thanks for comment, I will doublecheck it!